### DISCUSSION TITLE
Bug onResponseHandler() with ModbusClientTCPasync API

### DISCUSSION THREAD

**Discussion by dima-sz** [eModbus/eModbus#271] [Q&A]
Hi, was explore this library and found somthing that looks like a bug.

1. 
When ModbusClientTCPasync API - function onError can't return server_id so i started using onResponceHandler, but found strange behavior in difference to onError function: While onError can handle any type of Errors, onResponceHandler for some reason can't handle timeout responces and just do nothig until they see device back online. 
If i do somthing to create external errors like requesting wrong register - all works fine and i get response with indicated error. But when server device going offline - onResponceHandler just do nothing, while onErrorHandler immediately start to sent timeouts.

The more strange thing: if i do the same with ModbusClientRTU API - onError and onResponce work simmilary (with exception of server_id of course), no problems with timeouts and anything.

So quation is: how can i get timeout responces with server id (onResponce) while using TCPasync API. Is this a bug? or some undocumented difference beetwen TCP and RTU parts of library.

2. 
The next bug (or not he bug, but i think it is) against happen only on ModbusClientTCPasync API, and dosent replicate with RTU.
If client going restart, or just going online after a while and start requsting his server it will not send timeout error from devices that are offline. But if server device go online and offline back after client was restarted - timeout error are sended properly as intended. 
So for some reasone TCPasync just dont let restarted client to get timeerorrs from disconeted devices, while RTU can do it.

**Comment by Miq1** [comment]
Right, these are known bugs in the Async TCP handling. The first is fixed in the fix-async branch. Since there is currently more work to be done in the AsyncTCP code before it can be released (including fixing the second) to the master branch, I would suggest you using the fix-async branch for the time being.

@bertmelis Please take note.

**Reply by dima-sz** [reply]
Thank you, i will try it.
